app/models/project.rb

```diff
@@ -125,6 +125,20 @@ def last_deploy_by_group(before_time)
     releases.map { |group_id, deploys| [ group_id, deploys.sort_by(&:updated_at).last ] }.to_h
   end
 
+  def self.search_by_criteria(criteria)
+    scope = Project
+    scope = scope.search(criteria[:search]) if criteria[:search]
+    scope.order("#{sort_column(criteria[:sort])} #{sort_direction(criteria[:direction])}").page(criteria[:page])
+  end
+
+  def self.sort_column(column)
+    column_names.include?(column) ? column : "created_at"
```

Comment:
Not sure if hiding bad input is the way to go... if user wanted to sort by a field they typo'd they should probably get an error.